Maven Compiler Plugin a Dependency for the SDK

_Warning_: Github issues are not an official support channel for Recurly. If you have an urgent request we suggest you contact us through an official channel: support@recurly.com or https://recurly.zendesk.com

**Describe the bug**

I was surprised to see the `maven-complier-plugin` as a compiled dependency on the `recurly-client-java` sdk. While investigating why it was included as a dependency, I see it was introduced in relation to [a change attempting to move to Java 11 a few years ago](https://github.com/recurly/recurly-client-java/commit/bd2efbf1ced3fe4e5f6f200d04164a329d2281d8). It's not clear to me that it's still needed, or if it really ever was. Including the `maven-compiler-plugin` as an actual dependency on the SDK results in my Java 11 project failing to compile due to the following types of errors:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project internal: Compilation failure: Compilation failure:
[ERROR] /workspace-git/revenue/internal/src/main/java/com/company/revenue/guice/PaymentRouterModule.java:[12,8] com.company.revenue.PaymentRouterModule is not abstract and does not override abstract method configure() in com.google.inject.AbstractModule
```

**To Reproduce**

I have the following plugin already defined in my own pom:
```
<plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.8.1</version>
        <configuration>
          <source>11</source>
          <target>11</target>
          <compilerArgs>
            <arg>-XDcompilePolicy=simple</arg>
            <arg>-Xplugin:ErrorProne</arg>
          </compilerArgs>
          <annotationProcessorPaths>
            <path>
              <groupId>com.google.errorprone</groupId>
              <artifactId>error_prone_core</artifactId>
              <version>2.15.0</version>
            </path>
          </annotationProcessorPaths>
        </configuration>
      </plugin>
```

I also have dependencies on Google Guice, with example classes such as:

```
package com.sproutsocial.revenue.entitlement.guice;

...
import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
...

public class PaymentRouterModule extends AbstractModule {
  @Provides
  @Singleton
  public PaymentRouter providesPaymentRouter(
      @SystemDb Db db, MetricRegistry metrics, InternalConfiguration configuration) {
    return configuration.getPaymentRouterFactory().build(db, metrics);
  }
}

```

**Expected behavior**

I'd expect including the recurly SDK to not impact the rest of my project's compilation. Note excluding the maven-compiler-plugin from the recurly SDK dependency in my pom does resolve my compilation issue. I'll continue with that approach until there's a version with this dependency removed.


**Your Environment**

- Which version of this library are you using? `4.24.1`
- Which version of the language are you using? Java 11 with Google Guice

--- a/pom.xml
+++ b/pom.xml
@@ -61,9 +61,9 @@
         <java.version>1.8</java.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <okhttp3.version>4.7.2</okhttp3.version>
+        <okhttp3.version>4.12.0</okhttp3.version>
         <surefire.version>3.0.0-M4</surefire.version>
-        <jacoco.version>0.8.5</jacoco.version>
+        <jacoco.version>0.8.12</jacoco.version>
     </properties>
 
     <build>
@@ -214,16 +214,15 @@
                 <autoReleaseAfterClose>true</autoReleaseAfterClose>
               </configuration>
             </plugin>
+            <plugin>
+              <groupId>org.owasp</groupId>
+              <artifactId>dependency-check-maven</artifactId>
+              <version>9.1.0</version>
+            </plugin>
         </plugins>
     </build>
 
     <dependencies>
-        <dependency>
-            <groupId>org.apache.maven.plugins</groupId>
-            <artifactId>maven-compiler-plugin</artifactId>
-            <version>3.8.1</version>
-            <scope>compile</scope>
-        </dependency>
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
@@ -295,20 +294,13 @@
         </dependency>
 
         <!-- Specify versions of transitive dependencies
-            xerces:xercesImpl introduced through jacoco-maven-plugin and doxia-core
-              - can be removed when mvn dependency:list shows version 2.12.0 or higher and no snyk reported vulnerabilities
             plexus:plexus-utils introduced through jacoco-maven-plugin, maven-compiler-plugin, and others
               - can be removed when mvn dependency:list shows version 3.0.24 or higher and no snyk reported vulnerabilities
         -->
         <dependency>
-            <groupId>xerces</groupId>
-            <artifactId>xercesImpl</artifactId>
-            <version>2.12.2</version>
-        </dependency>
-        <dependency>
           <groupId>org.codehaus.plexus</groupId>
           <artifactId>plexus-utils</artifactId>
-          <version>3.2.1</version>
-      </dependency>
+          <version>3.5.1</version>
+        </dependency>
     </dependencies>
 </project>

